txstratum/healthcheck.py

```diff
@@ -0,0 +1,265 @@
+
+    async def get_health_check(self) -> ComponentHealthCheck:
+        """Return the fullnode health check status."""
+        health_check = ComponentHealthCheck(
+            component_name="fullnode",
+            component_type=ComponentType.FULLNODE,
+            # TODO: Ideally we should not use private fields. We'll fix this when fixing line 170
+            component_id=self.backend._base_url,
+            status=HealthCheckStatus.PASS,
+            time=datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ"),
+            output="fullnode is responding correctly",
+        )
+
+        try:
+            # TODO: We need to get the health information from the fullnode, but it's not implemented yet
+            await self.backend.version()
+
+            return health_check
+        except Exception:
+            health_check.status = HealthCheckStatus.FAIL
+            health_check.output = "couldn't connect to fullnode"
+
+            return health_check
+
+
+class MiningHealthCheck(ComponentHealthCheckInterface):
+    """
+    This class receives a manager instance as parameter and implements a health check method for it.
+
+    It will:
+    - Check that the manager has at least 1 miner
+    - Check that all 'tx_jobs' in the manager in the last 5 minutes have a status of done
+      and have a total_time lesser than 10 seconds.
+    - Check that all miners have submitted at least 1 job in the last 1 hour.
+      Note that subimitting a job doesn't mean solving a tx.
+
+    If at least one of the 'tx_jobs' has a 'total_time' of more than 10 seconds, the health check
+    will be returned as 'warn'.
+
+    If some of the 'tx_jobs' has status different than 'done', the health check will be returned
+    as 'fail'.
+
+    If there are no miners, the health check will be returned as 'fail'
+
+    If there are miners, but any of them has submitted a job in the last 1 hour, the health check
+    will be returned as 'fail'
+    """
+
+    def __init__(self, manager: "TxMiningManager") -> None:
+        """Init the class with the manager instance."""
+        self.manager = manager
+        self.last_manager_status = ComponentHealthCheck(
+            component_name="manager",
+            component_type=ComponentType.INTERNAL,
+            status=HealthCheckStatus.PASS,
+            output="everything is ok",
+        )
+
+    async def get_health_check(self) -> ComponentHealthCheck:
+        """Return the manager health check status."""
+        # Check that the manager has at least 1 miner
+        if not self.manager.has_any_miner():
+            return ComponentHealthCheck(
+                component_name="manager",
+                component_type=ComponentType.INTERNAL,
+                status=HealthCheckStatus.FAIL,
+                output="no miners connected",
+            )
+
+        # Check that all the miners submitted in the last 1 hour. If not, return failed.
+        if not self.manager.has_any_submitted_job_in_period(period=3600):
```

Comment:
Move 3600 to constant?

Reply:
Done in https://github.com/HathorNetwork/tx-mining-service/pull/104/commits/acc340e690cfd4bf508fa18b6b25a2a98385b5aa